bciers/apps/registration/tests/components/form/widgets/FileWidget.test.tsx

```diff
@@ -0,0 +1,134 @@
+import { userEvent } from "@testing-library/user-event";
+import { render, screen, waitFor } from "@testing-library/react";
+import { RJSFSchema } from "@rjsf/utils";
+import FormBase from "@/app/components/form/FormBase";
+import { actionHandler, useSession } from "@/tests/setup/mocks";
+import { Session } from "@/tests/setup/types";
+
+const fileFieldLabel = "FileWidget test field";
+const fileLabelRequired = `${fileFieldLabel}*`;
+export const dummyDataUri =
+  "data:application/pdf;name=dummy.pdf;base64,ZHVtbXk=";
+const mockFile = new File(["test"], "test.pdf", { type: "application/pdf" });
+
+vi.resetAllMocks();
+
+export const fileFieldSchema = {
+  type: "object",
+  required: ["fileTestField"],
+  properties: {
+    fileTestField: {
+      type: "string",
+      format: "data-url",
+      title: fileFieldLabel,
+    },
+  },
+} as RJSFSchema;
+
+export const fileFieldUiSchema = {
+  fileTestField: {
+    "ui:widget": "FileWidget",
+  },
+};
+
+describe("RJSF FileWidget", () => {
```

Comment:
Feels like this could be more thoroughly tested. What are we missing here?

Reply:
Okay I've done what I can here but some things to note as our FileWidget is super basic at the moment:
- There is no feature to remove the file. I added a test to verify that we can replace the current file though.
- The `accept` functionality for file types just blocks them from being selected in the document browser and we don't currently have any built in validation messages. I added a test that verifies that the input won't accept a non-pdf.
- I added a test to confirm that the `accept` option would accept .kml/.kmz. I wasn't sure if you meant you actually wanted that enabled for the file upload field on the operations form. It would be easy to do but I only added a test.
- Added a couple tests for the 20mb maximum``

Reply:
the test cases for FileWidget are very comprehensive now - great work @BCerki and @marcellmueller !